src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -540,6 +540,11 @@ export class TerminalService implements ITerminalService {
 	}
 
 	private async _onBeforeShutdownAsync(reason: ShutdownReason): Promise<boolean> {
+		if (this._backgroundedTerminalInstances.length) {
+			for (const terminal of this._backgroundedTerminalInstances.filter(b => !b.shouldPersist)) {
+				terminal.dispose();
```

Comment:
Just wondering whether this should be inside the try/catch?

Reply:
`dispose` won't throw so I think this is fine

Reply:
https://github.com/markw65/vscode/blob/05dce4918275b5e568a52c8c4e318f36dde0bf1f/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1200-L1205